### ISSUE TITLE
[Elixir] Implement new Concept Exercise: `with`

### ISSUE THREAD

**Issue by angelikatyborska** [exercism/v3#2305]
## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how the `with` special form works and when it is appropriate to use (e.g. to avoid nested `case`s)

## Learning objectives

- Know how to use `with` to chain matching clauses 
- Know how to use the `else` block in `with` to match on different kinds of errors
- Know when _not_ to use `with`

## Out of scope

?

## Concepts

- `with`

## Prerequisites

- `errors` (the `with` special form is most commonly used with error/ok-tuples and the exercise should teach that)
- `pattern-matching`
- other, depending on the story

## Resources to refer to

- https://hexdocs.pm/elixir/Kernel.SpecialForms.html#with/1
- https://elixir-lang.org/getting-started/mix-otp/docs-tests-and-with.html#with
- http://learningelixir.joekain.com/learning-elixir-with/
- https://openmymind.net/Elixirs-With-Statement/

## Analyzer

It will be necessary to use the analyzer to check that the `with` form was indeed used, and that no alternatives (e.g. `case`s) were used.

## Story

Look through existing stories (`/reference/stories/`) to find inspiration or an existing story that can work for this exercise.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by angelikatyborska** [comment]
Migrated to the Elixir track repo: https://github.com/exercism/elixir/issues/560
